XMLHttpRequest/send-redirect-to-cors.htm

```diff
@@ -13,56 +12,68 @@
-        var test = async_test(document.title + " (" + name + ")")
-        test.step(function() {
+      function redirect(code, name = code, method = "GET", body = null, explicitType = null, safelistContentType = false) {
+        async_test(t => {
           var client = new XMLHttpRequest()
-          client.onreadystatechange = function() {
-            test.step(function() {
-              if (client.readyState == 4) {
+          client.onreadystatechange = t.step_func(() => {
+            if (client.readyState == 4) {
+              if (explicitType !== "application/x-pony" || safelistContentType) {
                 assert_equals(client.status, 200);
-                assert_equals(client.getResponseHeader("x-request-method"),
-                              method);
+                assert_equals(client.getResponseHeader("x-request-method"), method);
                 var { body: expectedBody, type: expectedType } = extractBody(body);
-                if (setExplicitType) {
-                  expectedType = "application/x-pony";
+                if (explicitType !== null) {
+                  expectedType = explicitType
                 }
-                assert_equals(client.getResponseHeader("x-request-content-type"),
-                              expectedType);
-                assert_equals(client.getResponseHeader("x-request-data"),
-                              expectedBody);
-                test.done();
+                assert_equals(client.getResponseHeader("x-request-content-type"), expectedType);
+                assert_equals(client.getResponseHeader("x-request-data"), expectedBody);
+              } else {
+                // "application/x-pony" is not safelisted by corsenabled.py -> network error
+                assert_equals(client.status, 0)
+                assert_equals(client.statusText, "")
+                assert_equals(client.responseText, "")
+                assert_equals(client.responseXML, null)
               }
-            })
+              t.done();
+            }
+          })
+          let safelist = ""
+          if (safelistContentType) {
+            safelist = "?safelist_content_type"
           }
-          client.open(method, "resources/redirect.py?location="+encodeURIComponent("http://www2."+location.host+(location.pathname.replace(/[^\/]+$/, ''))+'resources/corsenabled.py')+"&code=" + code)
-          if (setExplicitType) {
-            client.setRequestHeader("Content-Type", "application/x-pony")
+          client.open(method, "resources/redirect.py?location="+encodeURIComponent("http://www2."+location.host+(location.pathname.replace(/[^\/]+$/, ''))+'resources/corsenabled.py')+safelist+"&code=" + code)
+          if (explicitType !== null) {
+            client.setRequestHeader("Content-Type", explicitType)
           }
           client.send(body)
-        })
+        }, document.title + " (" + name + ")")
       }
       redirect("301")
```

Comment:
Didn't you change behavior for this call?  It used to set the Content-Type header, but now no longer does.  Is that purposeful?